### ISSUE TITLE
Clarify explanations in the explainer

### ISSUE THREAD

**Issue by littledan** [rricard/proposal-refcollection#2]
We probably want to tweak the wording around garbage collection to make it more pedantic. In particular, engines don't 'release' things, but rather, if the symbol is impossible to forge, then the engine may reason that the value in the refm is unreachable, even if it's logically "never deleted".

Similarly, the text about object-ref stability could also use rewording for clarity, explaining that, at a high level, this is a never-changing map that you can just add entries to, and why this is nice to have (basically, it's dependable and meets use cases).

**Comment by sebmarkbage** [comment]
The explanation should start with why an indirection is preferable in the first place as opposed to a direct reference. I was expecting this to be solved with something similar to Rust's RefCell.

If a collection is needed maybe that's a reason to allow Symbols in WeakMap.

**Comment by rricard** [comment]
> I was expecting this to be solved with something similar to Rust's RefCell.

I started to fiddle with rust just recently, I'll make sure I take a look at RefCells as you're not the first one to mention them to me.

**Comment by rricard** [comment]
> If a collection is needed maybe that's a reason to allow Symbols in WeakMap.

That was my original idea, however we do not want to break the WeakMap does not take primitive types as keys property

**Comment by sebmarkbage** [comment]
Modeling this based on Symbols effectively put Symbols in a new category that is different from primitives by allowing it to contribute in garbage collection. It’s possible to have a Symbol that can go away and be impossible to recreate unlike other primitives. So I’d argue that Symbols can no longer be considered as the same kind of primitive as the others. Other primitives doesn’t make sense in WeakMap because they can always be recreated so they can never be collected. The rationale for excluding primitives in WeakMap doesn’t apply to Symbols.

We also general think that it’s ok to make (un)breaking changes to things that used to throw.

**Comment by ljharb** [comment]
@sebmarkbage see https://github.com/tc39/ecma262/issues/1194 for why allowing Symbols in WeakMap is effectively a nonstarter

**Comment by sebmarkbage** [comment]
@ljharb IMO the same arguments applies to this proposal and vice versa. Which is fine but that would also make this a non-starter if it exposes the Symbols as a mechanism to any api or hook. If it doesn’t then it shouldn’t use Symbols as a mechanism.

**Comment by ljharb** [comment]
Agreed.

**Comment by littledan** [comment]
It would be good to go into a bit more detail in the "behaviors" section. Why are these properties important, and how does the RefCollection proposal meet them?

**Comment by littledan** [comment]
@sebmarkbage I agree that it will be important to document this motivation about why "boxes" are represented this way and not like they are in several other languages. A cross-language comparison would be useful.

One piece to the motivation is compatibility with membranes. The RefCollection intermediates between Symbols and what they're referring to. Because it's used with object operations, Symbols + RefCollections are membrane-friendly--Records and Tuples can have Symbols in them and be primitive through and through, but if you want to get to what the Symbol is pointing to, you have to go through a RefCollection, which may be membrane-wrapped.

Overall, if we decided that Symbols could be WeakMap keys, then that would subsume the use case for this proposal, I agree.
